### ISSUE TITLE
Occassional failures of xt/nonblocking.t

### ISSUE THREAD

**Issue by choroba** [karupanerura/p5-AnyEvent-ForkManager#3]
When running `xt/nonblocking.t` 50 times on Linux, it finished with PASS 43 times, but 7 times with FAIL:

```
Interrupted system call at /home/choroba/perl5/lib/perl5/Test/SharedFork/Store.pm line 104.
Can't use an undefined value as a HASH reference at /home/choroba/perl5/lib/perl5/Test/SharedFork/Store.pm line 51.
END failed--call queue aborted at xt/nonblocking.t line 104.
```

**Comment by karupanerura** [comment]
Umm... Can you tell me version of your using `Test::SharedFork`and `Test::More`?
I think this issue is fixed newest version of `Test::SharedFork`and `Test::More`. maybe...

**Comment by karupanerura** [comment]
I see that this code is never failure.
```bash
while true; do
  prove -Ilib --failures xt/nonblocking.t
done
```

perl: v5.20.1
Test::More: 1.001014
Test::SharedFork: 0.3

**Comment by choroba** [comment]
I have the same version of the modules, but Perl 5.18.1.

**Comment by karupanerura** [comment]
It's fixed by: 72b675bbd9e3482d84ec388717b719190fb1a58c
